docs/tutorial/using-grpc-plugins.mdx

```diff
@@ -0,0 +1,552 @@
+  """
+  people: [Person!]!
+}
+```
+
+For this example, we won’t wrap the entire SWAPI, only the `people` resource and its endpoints.
+
+This schema has a single type, “Person”, and a query to get all the people or a specific person by ID.
+
+### Generate and Implement
+
+Now we can use `wgc` again to generate the Protobuf representation of our subgraph and boilerplate code to implement it in Golang.
+
+```bash
+wgc router plugin generate ./starwars
+```
+
+You’ll now see a few new files in the `generated` folder.
+
+```
+generated
+├── mapping.json
+├── service.pb.go
+├── service.proto
+├── service.proto.lock.json
+└── service_grpc.pb.go
+```
+
+In short, these files are generated helpers based on the schema we wrote. They help either translate GraphQL operations to gRPC or let you write type-safe resolvers in the plugin itself.
+
+
+<Note>
+  We recommend checking this folder into version control (e.g. Git).
+</Note>
+
+Now, let’s start implementing our resolvers. After generating, if you open `main.go`, you will see some errors about undefined types. These are remnants from the example schema’s resolver, and you can safely delete them to start from scratch.
+
+Here’s a starting point for implementing our new resolvers:
+
+```go
+package main
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"time"
+
+	service "github.com/wundergraph/cosmo/plugin/generated"
+
+	routerplugin "github.com/wundergraph/cosmo/router-plugin"
+	"google.golang.org/grpc"
+)
+
+const swapiBaseURL = "https://swapi.dev/api"
+
+func main() {
+	pl, err := routerplugin.NewRouterPlugin(func(s *grpc.Server) {
+		s.RegisterService(&service.StarwarsService_ServiceDesc, &StarwarsService{
+			client: &http.Client{
+				Timeout: 30 * time.Second,
+			},
+		})
+	})
+
+	if err != nil {
+		log.Fatalf("failed to create router plugin: %v", err)
+	}
+
+	pl.Serve()
+}
+
+type StarwarsService struct {
+	service.UnimplementedStarwarsServiceServer
+
+	client *http.Client
+}
+
+func (s *StarwarsService) QueryPeople(ctx context.Context, req *service.QueryPeopleRequest) (*service.QueryPeopleResponse, error) {
+	panic("not implemented")
+}
+```
+
+This updates the `main.go` that comes with the initial plugin template in a few ways:
+
+1. We removed resolvers for the old `QueryHello` RPC and a field `nextId` in the `StarwarsService` struct. We won't need it anymore.
+2. In the `StarwarsService` struct, we added a field of type `*http.Client`. 
+    - This will hold a persistent HTTP client that our endpoints can use for the whole lifetime of the plugin process.
+    - We set a 30-second request timeout. It’s important to set this for HTTP clients in Go because the default is unlimited and can lead to resource exhaustion from many pending requests.
+3. We add a constant `swapiBaseURL` that we'll later use to construct the URL for the SWAPI API endpoint.
+
+If you ran this now, the plugin would just panic (exit ungracefully) when we tried to use the `people` query. Let’s fix that.
```

Comment:
Explain why even it is super obvious

Reply:
Plugins have to be run by the router, I should switch this language to "if you published this now"

Reply:
I understand that, I think what confused me was that, this is not mentioned anywhere, so by saying `if you ran it now`, people might be tempted to see what kind of error they get, out of curiosity

Might just be me though

Reply:
I changed the phrasing and added a small explainer